processing/src/main/java/org/apache/druid/query/aggregation/first/DoubleFirstAggregatorFactory.java

```diff
@@ -256,20 +259,28 @@ public String getFieldName()
     return fieldName;
   }
 
+  @JsonProperty
+  public String getTimeColumn()
+  {
+    return timeColumn;
+  }
+
   @Override
   public List<String> requiredFields()
   {
-    return Arrays.asList(ColumnHolder.TIME_COLUMN_NAME, fieldName);
+    return Arrays.asList(timeColumn, fieldName);
   }
 
   @Override
   public byte[] getCacheKey()
   {
     byte[] fieldNameBytes = StringUtils.toUtf8(fieldName);
+    byte[] timeColumnBytes = StringUtils.toUtf8(timeColumn);
```

Comment:
nit: maybe a good opportunity to switch to `CacheKeyBuilder`? (same comment for all other aggs)

Reply:
yup, moved the aggs to `CacheKeyBuilder`